Tasks/Common/ios-signing-common/ios-signing-common.ts

```diff
@@ -472,12 +473,18 @@ export async function unlockKeychain(keychainPath: string, keychainPwd: string):
  * @param uuid
  */
 export async function deleteProvisioningProfile(uuid: string): Promise<void> {
-    let provProfilePath: string = getProvisioningProfilePath(uuid);
-    tl.warning('Deleting provisioning profile: ' + provProfilePath);
-    if (tl.exist(provProfilePath)) {
-        let deleteProfileCommand: ToolRunner = tl.tool(tl.which('rm', true));
-        deleteProfileCommand.arg(['-f', provProfilePath]);
-        await deleteProfileCommand.exec();
+    if (uuid) {
+        const provProfiles: string[] = tl.findMatch(getUserProvisioningProfilesPath(), uuid.trim() + '*');
+        if (provProfiles) {
+            provProfiles.forEach(async (provProfilePath) => {
```

Comment:
@brcrista considers foreach deprecated by `for..of`. I think using `for..of` is clearer.

Reply:
Will change in master.